File: Python/Quick_sort.py
def helpsort(array,start,end):
    temp_start=start
    if end-start>1:
        pivot=end
        while start is not pivot:
            #print(pivot,start, list(array))
            if array[pivot]>array[start]:
                start+=1
            elif array[pivot]<=array[start]:
                if pivot-1==start:
                    temp =array[pivot]
                else:
                    temp=array[pivot-1]
                    array[pivot-1]=array[pivot]
                array[pivot]=array[start]
                array[start]=temp
                pivot-=1
        #print(list(array))
        halfsort(array,pivot+1,end)
        halfsort(array,temp_start,pivot-1)
        return array
    else:
        return array
def quicksort(array):
    return helpsort(array,0,len(array)-1)

test = [21, 4, 1, 3, 9, 20, 25, 6, 21, 14]
print quicksort(test)

File: C++/Quick_Sort.cpp
#include <iostream>
using namespace std;

void swap(int arr[] , int pos1, int pos2){
	int temp;
	temp = arr[pos1];
	arr[pos1] = arr[pos2];
	arr[pos2] = temp;
}

int partition(int arr[], int low, int high, int pivot){
	int i = low;
	int j = low;
	while( i <= high){
		if(arr[i] > pivot){
			i++;
		}
		else{
			swap(arr,i,j);
			i++;
			j++;
		}
	}
	return j-1;
}

void quickSort(int arr[], int low, int high){
	if(low < high){
	int pivot = arr[high];
	int pos = partition(arr, low, high, pivot);

	quickSort(arr, low, pos-1);
	quickSort(arr, pos+1, high);
	}
}

int main()
{
	int n ;
	cout << " Enter the size of array :"<<endl;
	cin>>n;
	int arr[n];
	cout<<"Enter the number of elements"<<endl;
	for( int i = 0 ; i < n; i++){
		cin>> arr[i];
	}
	quickSort(arr, 0 , n-1);
	cout<<"The sorted array are: ";
	for( int i = 0 ; i < n; i++){
		cout<< arr[i]<<"\t";
	}

}
